x-pack/plugins/screenshotting/server/formats/pdf/index.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { groupBy } from 'lodash';
+import type { Logger } from 'src/core/server';
+import { LayoutParams, LayoutTypes } from '../../../common';
+import { ScreenshotResult, ScreenshotOptions } from '../../screenshots';
+import { FormattedScreenshotResult } from '../types';
+import { pngsToPdf } from './pdf_maker';
+
+const supportedLayouts = [LayoutTypes.PRESERVE_LAYOUT, LayoutTypes.CANVAS, LayoutTypes.PRINT];
+
+/**
+ * PDFs can be a single, long page or they can be multiple pages. For example:
+ *
+ * => When creating a PDF intended for print multiple PNGs will be spread out across pages
+ * => When creating a PDF from a Canvas workpad, each page in the workpad will be placed on a separate page
+ */
+export type PdfLayoutParams = LayoutParams<typeof supportedLayouts[number]>;
+
+/**
+ * Options that should be provided to a PDF screenshot request.
+ */
+export interface PdfScreenshotOptions extends ScreenshotOptions<'pdf'> {
+  title?: string;
+  logo?: string;
+  /**
+   * We default to the "print" layout if no ID is specified for the layout
+   */
+  layout: PdfLayoutParams;
+}
+
+/**
+ * Final, formatted PDF result
+ */
+export interface PdfScreenshotResult extends Omit<FormattedScreenshotResult, 'results'> {
+  metadata: { pageCount: number };
```

Comment:
Can we move that into the metrics interface? It was impossible to put `pageCount` into the metrics before this change, but now it can be added there.